six/modules/python/cphd/tests/test_make_cphd.py

```diff
@@ -0,0 +1,738 @@
+    ep.monostatic.posVelErr.p2 = 1.0
+    ep.monostatic.posVelErr.p3 = 1.0
+    ep.monostatic.posVelErr.v1 = 1.0
+    ep.monostatic.posVelErr.v2 = 1.0
+    ep.monostatic.posVelErr.v3 = 1.0
+
+    ep.monostatic.posVelErr.corrCoefs = six.makeScopedCopyableCorrCoefs()
+    for s1, idx1, s2, idx2 in itertools.product(['p', 'v'],
+                                                [1, 2, 3],
+                                                ['p', 'v'],
+                                                [1, 2, 3]):
+        name = '{}{}{}{}'.format(s1, idx1, s2, idx2)
+        if name in dir(ep.monostatic.posVelErr.corrCoefs):
+            ep.monostatic.posVelErr.corrCoefs.__setattr__(name, 0.8)
+
+    ep.monostatic.posVelErr.positionDecorr.corrCoefZero = 0.5
+    ep.monostatic.posVelErr.positionDecorr.decorrRate = 1.0
+
+    ep.monostatic.radarSensor.rangeBias = 0.5
+    ep.monostatic.radarSensor.clockFreqSF = 1.0
+    ep.monostatic.radarSensor.collectionStartTime = 1.0
+    ep.monostatic.radarSensor.rangeBiasDecorr = \
+        cphd.makeScopedCopyableDecorrType()
+    ep.monostatic.radarSensor.rangeBiasDecorr.corrCoefZero = 0.5
+    ep.monostatic.radarSensor.rangeBiasDecorr.decorrRate = 1.0
+
+    ep.monostatic.tropoError = six.makeScopedCopyableTropoError()
+    ep.monostatic.tropoError.tropoRangeVertical = 5.0
+    ep.monostatic.tropoError.tropoRangeSlant = 5.0
+    ep.monostatic.tropoError.tropoRangeDecorr.corrCoefZero = 0.5
+    ep.monostatic.tropoError.tropoRangeDecorr.decorrRate = 1.0
+
+    ep.monostatic.ionoError = six.makeScopedCopyableIonoError()
+    ep.monostatic.ionoError.ionoRangeVertical = 5.0
+    ep.monostatic.ionoError.ionoRangeRateVertical = 5.0
+    ep.monostatic.ionoError.ionoRgRgRateCC = 0.5
+    ep.monostatic.ionoError.ionoRangeVertDecorr.corrCoefZero = 0.5
+    ep.monostatic.ionoError.ionoRangeVertDecorr.decorrRate = 1.0
+
+    cphd.errorParameters = ep
+
+    # ProductInfo block
+
+    pi = cphd.makeScopedCopyableProductInfo()
+    pi.profile = 'Profile'
+
+    ci = cphd.CreationInfo()
+    ci.application = 'Application'
+    ci.datetime = six.DateTime()
+    ci.site = 'Area51'
+    for args in [('Param1', 'Value1'),
+                 ('Param1', 'Value2')]:
+        ci.parameter.push_back(get_parameter(*args))
+    pi.creationInfo.append(ci)
+
+    for args in [('Param1', 'Value1')]:
+        pi.parameter.push_back(get_parameter(*args))
+
+    metadata.productInfo = pi
+
+    # GeoInfo block (TODO)
+
+    # MatchInfo block (TODO)
+
+    xml_parser = cphd.CPHDXMLControl()
+
+    try:
+        xml_str = xml_parser.toXMLString(metadata)
+        meta_from_str = xml_parser.fromXMLString(xml_str)
+        xml_str_from_meta_from_str = xml_parser.toXMLString(meta_from_str)
+        # TODO: Verify round-tripped strings are identical. Because of the
+        # use of std::unordered_map this is not guaranteed at the moment
+        print('Test passed')
+    except Exception:
+        print('Test failed')
```

Comment:
missing a newline if we care about being pep8 compliant

Reply:
`pycodestyle` complains when I add an extra blank line at the end of file.

Reply:
my bad, I think the print('Test failed') you have is actually "print('Test failed')\n" if you were to display carriage returns, which is compliant